core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -165,17 +169,41 @@ $(document).ready(function() {
         var completedColumnName = 'completed';
         var durationColumnName = 'duration';
         var conf = {
+          "data": array,
           "columns": [
-            {name: 'version'},
-            {name: 'appId', type: "appid-numeric"},
-            {name: 'appName'},
-            {name: attemptIdColumnName},
-            {name: startedColumnName},
-            {name: completedColumnName},
-            {name: durationColumnName, type: "title-numeric"},
-            {name: 'user'},
-            {name: 'lastUpdated'},
-            {name: 'eventLog'},
+            {name: 'version', data: 'version' },
+            {
+              name: 'appId', 
+              type: "appid-numeric", 
+              data: 'id',
+              render:  (id, type, row) => `<span title="${id}"><a href="${row.attemptUrl}">${id}</a></span>`
+            },
+            {name: 'appName', data: 'name' },
+            {
+              name: attemptIdColumnName, 
+              data: 'attemptId',
+              render: (attemptId, type, row) => (attemptId ? `<a href="${row.attemptUrl}">${attemptId}</a>` : '')
+            },
+            {name: startedColumnName, data: 'startTime' },
+            {name: completedColumnName, data: 'endTime' },
+            {name: durationColumnName, type: "title-numeric", data: 'duration' },
+            {name: 'user', data: 'sparkUser' },
+            {name: 'lastUpdated', data: 'lastUpdated' },
+            {
+              name: 'eventLog', 
+              data: 'log', 
+              render: (log, type, row) => `<a href="${log}" class="btn btn-info btn-mini">Download</a>` 
+            },
+          ],
+          "aoColumnDefs": [
```

Comment:
@srowen 
This part is the logic, not in the template anymore.

Here what I am trying to do is to change the style of first 3 columns only if hasMultipleAttemptsit will be set to white background.

Reply:
OK I see. Does the table look the exact same before and after - there's no visual change? if so I don't think we need screenshots.

Reply:
Yeah it looks similar, but I will post some photos from prod/dev history servers from my side.

Reply:
Before
<img width="1771" alt="dev" src="https://user-images.githubusercontent.com/14618484/104816576-1739e000-5857-11eb-8f39-a2bb9d5a4085.png">

After
<img width="1763" alt="prod" src="https://user-images.githubusercontent.com/14618484/104816581-1acd6700-5857-11eb-8062-d35551d609f4.png">